src/views/TrackPositionAnimation/TrackPositionAnimationView.tsx

```diff
@@ -164,14 +147,22 @@ const TrackPositionAnimationView: FunctionComponent<TrackPositionAnimationProps>
         })
     }, [dataFrames])
 
+    const { focusTime, setTimeFocus } = useTimeFocus()  // state imported from recording context
+    const getTimeFromFrame = useCallback((frame: PositionFrame | undefined) => frame?.timestamp ?? -1, [])
+
+    const { handleOutsideTimeUpdate, handleFrameTimeUpdate } = useLiveTimeSyncing(setTimeFocus, animationState, animationStateDispatch, getTimeFromFrame)
+    useEffect(() => handleOutsideTimeUpdate(focusTime), [handleOutsideTimeUpdate, focusTime])
+    useEffect(() => handleFrameTimeUpdate(), [handleFrameTimeUpdate])
+
+    useTimeWindowSyncing(animationState, animationStateDispatch, getTimeFromFrame)
+
     const currentProbabilityFrame = useMemo(() => {
-        const linearFrame = dataFrames[animationState.currentFrameIndex].decodedPositionFrame
+        const linearFrame = animationState.frameData[animationState.currentFrameIndex]?.decodedPositionFrame
```

Comment:
`dataFrames` comes from the input data, while `animationState.frameData` is the complete data set that's currently persisted in the state.

If we ever manage to get buffered loading (or other ways of incrementally building the state frames) set up, the distinction will be important.